assets/js/components/TourTooltip.js

```diff
@@ -0,0 +1,136 @@
+/**
+ * TourTooltip component.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import Card, { CardActions } from '@material/react-card';
+import classnames from 'classnames';
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { createIncrementalArrayBySize } from '../util/create-incremental-array-by-size';
+import CloseIcon from '../../svg/close.svg';
+import Button from './Button';
+
+export default function TourTooltip( {
+	backProps,
+	closeProps,
+	index,
+	isLastStep,
+	primaryProps,
+	size,
+	step,
+	tooltipProps,
+} ) {
+	const indicatorArray = size > 1 ? createIncrementalArrayBySize( size ) : [];
+	const getIndicatorClassName = ( indicatorIndex ) => classnames( 'googlesitekit-tooltip-indicator', {
+		active: indicatorIndex === index,
+	} );
+
+	const gotItText = __( 'Got it', 'google-site-kit' );
+
+	return (
+		<div className="googlesitekit-tour-tooltip" { ...tooltipProps }>
+			<Card className="googlesitekit-tooltip-card">
+				<div className="googlesitekit-tooltip-body">
+					<h2 className="googlesitekit-tooltip-title">
+						{ step.title }
+					</h2>
+					<div className="googlesitekit-tooltip-content">
+						{ step.content }
+					</div>
+				</div>
+				<CardActions className="googlesitekit-tooltip-actions">
+					<ul className="googlesitekit-tooltip-indicators">
+						{ indicatorArray.map( ( indicatorIndex ) => (
+							<li
+								key={ `indicator-${ indicatorIndex }` }
+								className={ getIndicatorClassName( indicatorIndex ) }
+							/>
+						) ) }
+					</ul>
+					<div className="googlesitekit-tooltip-buttons">
+						{ index !== 0 && (
+							<Button
+								className="googlesitekit-tooltip-button"
+								text
+								{ ...backProps }
+							>
+								{ __( 'Back', 'google-site-kit' ) }
+							</Button>
+						) }
+						{ ! isLastStep && (
+							<Button
+								className="googlesitekit-tooltip-button"
+								text
+								{ ...primaryProps }
+							>
+								{ __( 'Next', 'google-site-kit' ) }
+							</Button>
+						) }
+						{ isLastStep && (
+							<Button
+								className="googlesitekit-tooltip-button"
+								text
+								{ ...closeProps }
+								aria-label={ gotItText }
+								title={ gotItText }
```

Comment:
@caseymorrisus What's the reason for making this an `aria-label` and `title` here, but not do the same for the other buttons? Or, why is that even necessary here? Since the button has a human-readable text as content, why the extra props/attributes here?

Reply:
@felixarntz I agree that it seems like overkill (and may even be bad a11y) but wanted to be consistent with the `react-joyride` library and its other buttons. Props provided by `react-joyride` that are being spread (`closeProps`) include `aria-label` and `title` values. The reason this button is the only one to have overrides is because `closeProps` prop is trying to set the `aria-label` to `Close` (as well as `title`) for the button that says `Got it` as I'm re-using the rest of the functionality provided.

Reply:
Leaving this here after our Slack message exchange: We'll need to ensure no user-facing strings come from Joyride defaults, since otherwise they wouldn't be translated via WordPress.